src/test/java/com/andersen/twitter/UserServiceTests.java

```diff
@@ -0,0 +1,157 @@
+package com.andersen.twitter;
+
+import com.andersen.twitter.entities.User;
+import com.andersen.twitter.repositories.UserRepository;
+import com.andersen.twitter.service.TweetService;
+import com.andersen.twitter.service.UserService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+public class UserServiceTests {
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private TweetService tweetService;
+
+    @InjectMocks
+    private UserService userService;
+
+    private User user;
+
+    @BeforeEach()
+    public void setup() {
+        user = new User( "TwitterAdmin", "Twitter Admin", "dog.png");
+    }
+
+    @Test
+    public void CreateUser_ValidUser_UserDaoCreateCalled() {
+        when(userRepository.save(user)).thenReturn(user);
+        User newuser = userService.createUser(user);
+        assertEquals(user, newuser);
+    }
+
+    @Test
+    public void CreateUser_NullUser_NullPointerExceptionThrown() {
+        when(userRepository.save(null)).thenThrow(new NullPointerException());
+        Exception exception = assertThrows(NullPointerException.class, () -> {
+            userService.createUser(null);
+        });
+        String expectedException = "java.lang.NullPointerException";
+        String actualException = exception.getClass().getName();
+
+        assertEquals(expectedException,actualException);
+    }
+
+    @Test
+    public void CreateUser_NullUserNullRepository_NullPointerExceptionThrown() {
+        userService.setUserRepository(null);
+        Exception exception = assertThrows(NullPointerException.class, () -> {
+            userService.createUser(null);
+        });
+        String expectedException = "java.lang.NullPointerException";
+        String actualException = exception.getClass().getName();
+
+        assertEquals(expectedException,actualException);
+    }
+
+    @Test
+    public void FindUser_ValidUsername_NotNullUserReturned() {
+        when(userRepository.findByUsername("TwitterAdmin")).thenReturn(user);
+        assertEquals(user, userService.findUser("TwitterAdmin"));
```

Comment:
better to create a local variable with the result of method call. It'll help you to avoid an additional work in case you'll have more checks in the future for this method